libkbfs/md_journal.go

```diff
@@ -843,6 +863,29 @@ func (j mdJournal) length() (uint64, error) {
 	return j.j.length()
 }
 
+func (j mdJournal) atLeastNNonLocalSquashes(
+	numNonLocalSquashes uint64) (bool, error) {
+	latestRev, err := j.readLatestRevision()
+	if err != nil {
+		return false, err
+	}
+
+	_, entries, err := j.j.getEntryRange(
+		latestRev-MetadataRevision(numNonLocalSquashes), latestRev)
+	if err != nil {
+		return false, err
+	}
+
+	count := uint64(0)
+	for _, e := range entries {
+		if !e.IsLocalSquash {
```

Comment:
Since you're grabbing the last entries `numNonLocalSquashes`, you can return `false` immediately once you encounter an entry with `e.IsLocalSquash`, right?

But if all the local squashes are a prefix of the journal, can't you just look up `latestRev - numNonLocalSquashes`, and return `!e.IsLocalSquash`?